core/primitives/src/runtime/parameter_table.rs

```diff
@@ -149,23 +146,23 @@ impl ParameterTable {
         Ok(())
     }
 
-    fn json_map(
+    fn yaml_map(
         &self,
         params: impl Iterator<Item = &'static Parameter>,
         remove_prefix: &'static str,
-    ) -> serde_json::Value {
-        let mut json = serde_json::Map::new();
+    ) -> serde_yaml::Value {
+        let mut yaml = serde_yaml::Mapping::new();
         for param in params {
             let mut key: &'static str = param.into();
             key = key.strip_prefix(remove_prefix).unwrap_or(key);
             if let Some(value) = self.get(*param) {
-                json.insert(key.to_owned(), value.clone());
+                yaml.insert(key.into(), value.clone());
             }
         }
-        json.into()
+        yaml.into()
     }
```

Comment:
BTW: This whole construct to group parameters by a prefix should probably be revisited, too. When we wanted to make it a linear list of parameters, this was the least ugly work-around we found. (But not part of this PR.)

Reply:
I agree that we should simplify this, especially given that there is only one usage to construct `VMLimitConfig` above. I haven't found a simple way to do this that doesn't require macro programming (as this is basically what serde::Deserialize that we use now relies on), but I'll keep looking.